pkg/ensurance/analyzer/analyzer.go

```diff
@@ -187,332 +187,161 @@ func (s *AnormalyAnalyzer) Analyze(state map[string][]common.TimeSeries) {
+					threshold = true
+					impacted = append(impacted, influced...)
 				}
 			} else {
-				// node metrics
-				klog.Infof("PodName or PodNamespace is empty")
 				threshold = true
+				return threshold, []types.NamespacedName{}
 			}
 		}
+	}
+	return threshold, impacted
+}
 
-		klog.V(4).Infof("DoAnalyze: key %s, threshold %v", key, threshold)
-
-		//step3: check is triggered action or restored, set the detection
-		if threshold {
-			s.restored[key] = 0
-			triggered := utils.GetUint64FromMaps(key, s.triggered)
-			triggered++
-			s.triggered[key] = triggered
-			if triggered >= uint64(utils.GetInt32withDefault(object.AvoidanceThreshold, known.DefaultAvoidedThreshold)) {
-				dc.Triggered = true
-				dc.BeInfluencedPods = impacted
-			}
-		} else {
-			s.triggered[key] = 0
-			restored := utils.GetUint64FromMaps(key, s.restored)
-			restored++
-			s.restored[key] = restored
-			if restored >= uint64(utils.GetInt32withDefault(object.RestoreThreshold, known.DefaultRestoredThreshold)) {
-				dc.Restored = true
-			}
-		}
+func (s *AnormalyAnalyzer) analyze(key string, object ensuranceapi.ObjectiveEnsurance, stateMap map[string][]common.TimeSeries) (ecache.DetectionCondition, error) {
+	var dc = ecache.DetectionCondition{Strategy: object.Strategy, ObjectiveEnsuranceName: object.Name, ActionName: object.AvoidanceActionName}
 
-	} else {
+	state, ok := stateMap[object.MetricRule.Name]
+	if !ok {
+		return dc, fmt.Errorf("metric %s not found", object.MetricRule.Name)
+	}
 
-		//step1: get metric value
-		value, err := s.getMetricFromMap(object.MetricRule.Name, state, object.MetricRule.Selector)
-		if err != nil {
-			return dc, err
-		}
+	//step1: get series from value
+	series, err := s.getSeries(state, object.MetricRule.Selector, object.MetricRule.Name)
+	if err != nil {
+		return dc, err
+	}
 
-		//step2: use opa to check if triggered
-		triggered := s.evaluator.EvalWithMetric(object.MetricRule.Name, float64(object.MetricRule.Value.Value()), value)
+	//step2: use opa to check if triggered and get impacted pods for container MetricRule
+	threshold, impacted := s.trigger(series, object)
 
-		//step3: check is triggered action or restored, set the detection
-		if triggered {
-			s.restored[key] = 0
-			triggered := utils.GetUint64FromMaps(key, s.triggered)
-			triggered++
-			s.triggered[key] = triggered
-			if triggered >= uint64(utils.GetInt32withDefault(object.AvoidanceThreshold, known.DefaultAvoidedThreshold)) {
-				dc.Triggered = true
-			}
-		} else {
-			s.triggered[key] = 0
-			restored := utils.GetUint64FromMaps(key, s.restored)
-			restored++
-			s.restored[key] = restored
-			if restored >= uint64(utils.GetInt32withDefault(object.RestoreThreshold, known.DefaultRestoredThreshold)) {
-				dc.Restored = true
-			}
-		}
-	}
+	klog.V(4).Infof("DoAnalyze: key %s, threshold %v", key, threshold)
+
+	//step3: check is triggered action or restored, set the detection
+	s.actionTriggeredRestored(threshold, key, object, &dc, impacted)
```

Comment:
getSeries,trigger, actionTriggeredRestored 

 the funtion name style must be consistent